src/http/request.cr

```diff
@@ -34,8 +35,8 @@ class HTTP::Request
     return unless request_line
 
     method, path, http_version = request_line.split
-    HTTP.parse_headers_and_body(io) do |headers, body|
-      return new method, path, headers, body.try &.read, http_version
+    HTTP.parse_headers_and_body(io) do |headers, cookies, body|
+      return new method, path, headers, body.try &.read, http_version, cookies
```

Comment:
Similarly maybe we could have something like `headers, cookies = Cookies.extract_from_headers(headers)` here?